lib/Sema/MiscDiagnostics.cpp

```diff
@@ -3455,15 +3458,29 @@ static void diagnoseUnintendedOptionalBehavior(TypeChecker &TC, const Expr *E,
       size_t optionalityDifference = 0;
       if (!isOptionalToAnyCoercion(srcType, destType, optionalityDifference))
         return;
-
-      TC.diagnose(subExpr->getStartLoc(), diag::optional_to_any_coercion,
-                  /* from */ srcType, /* to */ destType)
+      
+      // If we're implicitly unwrapping from IUO to Any then emit a custom
+      // diagnostic
+      if (hasImplicitlyUnwrappedResult(subExpr)) {
+        if (auto decl = getDeclForImplicitlyUnwrappedExpr(subExpr)) {
+          TC.diagnose(subExpr->getStartLoc(), diag::iuo_to_any_coercion,
+                      /* from */ srcType, /* to */ destType)
+          .highlight(subExpr->getSourceRange());
```

Comment:
LGTM! Could you please run `clang-format` on all of these changes, they are not currently formatted at the moment.

Reply:
I did run it but not sure why it didn't indent this properly 🤷‍♂️ Anyway, it's fixed now!